capstone/capapi/views.py

```diff
@@ -56,19 +57,111 @@ class CitationViewSet(viewsets.GenericViewSet, mixins.RetrieveModelMixin, mixins
 
 
 class CaseViewSet(viewsets.GenericViewSet, mixins.RetrieveModelMixin, mixins.ListModelMixin,):
-    """
-    Browse all cases
-    """
     serializer_class = serializers.CaseSerializer
     http_method_names = ['get']
+    lookup_field = 'slug'
     queryset = models.CaseMetadata.objects.all()
     renderer_classes = (renderers.BrowsableAPIRenderer, renderers.JSONRenderer)
-    lookup_field = 'slug'
+    permission_classes = (permissions.IsAPIUser,)
     filter_class = filters.CaseFilter
 
+    def download_many(self):
+        cases = self.queryset.all().order_by('decision_date')
+
+        for backend in list(self.filter_backends):
+            cases = backend().filter_queryset(self.request, cases, self)
+
+        # get max number of cases to download from query
+        # set max number to daily max if unspecified
+        cases = cases[:int(self.request.query_params.get('max', settings.API_CASE_DAILY_ALLOWANCE))]
```

Comment:
I think this line might be replaced by pagination?

Reply:
Actually the thought here was to allow the user to specify how many cases they want, instead of getting all possible for a certain query. Something like, get me the first case, or ten cases.

Reply:
That makes sense, but I think they can do it through pagination. Not sure how you have pagination configured, but it would be something like `http://api.example.org/accounts/?page=4&page_size=100`. So `http://api.example.org/accounts/` will get the first 100, `http://api.example.org/accounts/?page_size=10` will get the first 10, and so on. It's a bit less intuitive, but since you have to figure out pagination to work with any API anyway I think it's usable enough.

(Note that `paginate_queryset` is implemented the same as your line here -- it does a `query = query[start:end]`, which I'm not sure how the Django ORM will interpret the second time. The code now asks for two limit clauses in the sql query, which doesn't make sense. So if we do want to keep both we have to figure out how they should interact.)

Reply:
Aha! Yes. Changed to using DRF's LimitOffsetPagination — I find "limit" to be much more intuitive than "page size".